lib/phoenix/live_dashboard/pages/ecto_stats_page.ex

```diff
@@ -3,51 +3,112 @@ defmodule Phoenix.LiveDashboard.EctoStatsPage do
   use Phoenix.LiveDashboard.PageBuilder
   import Phoenix.LiveDashboard.Helpers
 
-  @compile {:no_warn_undefined, [Decimal, EctoPSQLExtras]}
+  @compile {:no_warn_undefined, [Decimal, EctoPSQLExtras, {Ecto.Repo, :all_running, 0}]}
   @disabled_link "https://hexdocs.pm/phoenix_live_dashboard/ecto_stats.html"
+  @page_title "Ecto Stats"
 
   @impl true
-  def init(%{repo: nil}), do: {:ok, %{repo: nil, ecto_options: []}}
+  def init(%{repos: repos, ecto_psql_extras_options: ecto_options}) do
+    capabilities = for repo <- List.wrap(repos), do: {:process, repo}
+    repos = repos || :auto_discover
 
-  def init(%{repo: repo, ecto_psql_extras_options: ecto_options}),
-    do: {:ok, %{repo: repo, ecto_options: ecto_options}, process: repo}
+    {:ok, %{repos: repos, ecto_options: ecto_options}, capabilities}
+  end
 
   @impl true
-  def mount(_params, %{repo: repo, ecto_options: ecto_options}, socket) do
-    {:ok,
-     assign(socket,
-       repo: repo,
-       ecto_options: ecto_options,
-       info_module: info_module_for(repo)
-     )}
+  def mount(_params, %{repos: repos, ecto_options: ecto_options}, socket) do
+    socket = assign(socket, ecto_options: ecto_options)
+
+    fun =
+      case repos do
+        :auto_discover ->
+          fn _ -> auto_discover() end
+
+        repos when is_list(repos) ->
+          &filter_repos_with_extra/1
+      end
+
+    case fun.(repos) do
+      {:ok, repos} ->
+        {:ok, assign(socket, :repos, repos)}
+
+      {:error, error} ->
+        {:ok, assign(socket, :error, error)}
+    end
+  end
+
+  defp auto_discover do
+    with true <- function_exported?(Ecto.Repo, :all_running, 0),
+         [_ | _] = repos <- Ecto.Repo.all_running(),
+         [_ | _] = named_repos <- Enum.filter(repos, &is_atom/1),
+         [_ | _] = repos_with_extra <- Enum.filter(named_repos, &extra_available?/1) do
```

Comment:
Maybe have this:

    [_ | _] = repos <- named_stats_available_repos()

And then:

    defp named_stats_available_repos do
      Enum.filter(Ecto.Repo.all_running(), &is_atom(&1) and extra_available?(&1))
    end

Perhaps even move the is_atom check to inside extra_available?, since you are calling `repo.__adapter__` in there.